testlapack/dlaqr5.go

```diff
@@ -0,0 +1,151 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"log"
+	"testing"
+
+	"github.com/gonum/floats"
+)
+
+type Dlaqr5er interface {
+	Dlaqr5(wantt, wantz bool, kacc22 int, n, ktop, kbot, nshfts int, sr, si []float64, h []float64, ldh int, iloz, ihiz int, z []float64, ldz int, v []float64, ldv int, u []float64, ldu int, nh int, wh []float64, ldwh int, nv int, wv []float64, ldwv int)
+}
+
+func Dlaqr5Test(t *testing.T, impl Dlaqr5er) {
+	r, err := zip.OpenReader("../internal/testdata/dlaqr5test/dlaqr5data.zip")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+		tc, err := f.Open()
+		if err != nil {
+			log.Fatal(err)
+		}
+		n, nshfts, ktop, kbot, sr, si, hOrig, hwant, zwant := readCase(tc)
+		tc.Close()
+
+		v := make([]float64, nshfts/2*3)
+		u := make([]float64, (3*nshfts-3)*(3*nshfts-3))
+		nh := n
+		wh := make([]float64, (3*nshfts-3)*n)
+		nv := n
+		wv := make([]float64, n*(3*nshfts-3))
+
+		for _, ldh := range []int{n, n + 1, n + 10} {
+			h := make([]float64, (n-1)*ldh+n)
+			for _, kacc22 := range []int{0, 1, 2} {
+				copySquareMatrix(n, h, ldh, hOrig)
+				z := eye(n, ldh)
+
+				impl.Dlaqr5(true, true, kacc22,
+					n, ktop, kbot,
+					nshfts, sr, si,
+					h, ldh,
+					0, n-1, z, ldh,
+					v, 3,
+					u, 3*nshfts-3,
+					nh, wh, nh,
+					nv, wv, 3*nshfts-3)
+
+				if !equalApprox(n, h, ldh, hwant, 1e-14) {
```

Comment:
The results are not bitwise equal with fortran. For bigger matrices (starting at around n=20) the tolerance must be increased to 1e-13. I don't know whether it indicates a bug or not.